### DISCUSSION TITLE
How do you catch exceptions?

### DISCUSSION THREAD

**Discussion by gunsuka** [php-mime-mail-parser/php-mime-mail-parser#386] [Q&A] [answered]
I started using this parser to parse email.  It works good.

However, sometimes I can get an error like this:

Warning</b>:  input is not rfc822 compliant: unexpected end of header in <b>/var/www/site/composer/vendor/php-mime-mail-parser/php-mime-mail-parser/src/Parser.php</b> on line <b>459</b><br />

I feed a message into the parser and then perform this:

$msg['arrayHeaderTo'] = $parser->getAddresses('to');

I have not bothered to look at the source, I'm sure it is bad.  I'm trying to catch the exception.  I have tried putting it into a try / catch but no exception is triggered.

I can ignore it with @$parser->setText($messageContent);   but that is not cool either.

What is the proper way to catch exceptions?

**Comment by eXorus** [comment]
Yes you can't catch a warning because it's not an exception.

I think the warning is raised by the method mailparse_rfc822_parse_addresses().

Could you share the header `to` to investigate more?

**Comment by gunsuka** [comment]
It seems to be getting the warning when trying to pull the TO data.  This is the TO header:

To: "=?UTF-8?B?Q29udGFiaWxpdMOg?= DVi Svc S.r.l.\"    <ita@example.net>"@relay.ccc.it

**Reply by gunsuka** [reply]
Oh!  I just realized that the string that Git is showing is NOT what I posted!    I have put the TO header in a text file and am going to attach it and see if it makes it without being altered.

Essentially, what has been modified in my posting is after S.r.l. there is a backslash \ that is escaping the "


[to.txt](https://github.com/php-mime-mail-parser/php-mime-mail-parser/files/8210086/to.txt)

**Reply by eXorus** [reply, marked as answer]
So I confirm it's an issue on mailparse not php-mime-mail-parser but it's only a warning not an error so the process is not stopped.
<img width="932" alt="Capture d’écran 2022-03-10 à 13 10 27" src="https://user-images.githubusercontent.com/1255561/157659070-1ef01b7b-7aca-46a9-a95c-79f607ccb18e.png">

There is something wrong what is the expected result?
Raw: `"=?UTF-8?B?Q29udGFiaWxpdMOg?= DVi Svc S.r.l.\"    <ita@example.net>"@relay.ccc.it`
iconv_mime_decode: `"Contabilità DVi Svc S.r.l.\"    <ita@example.net>"@relay.ccc.it`

Why do we have an `\"`?
